libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java

```diff
@@ -822,33 +829,54 @@ public void queryAsString(StringBuilder resultBuilder, QuerySpec querySpec, int
 			Cursor cursor = null;
 			try {
 				cursor = DBHelper.getInstance(db).limitRawQuery(db, sql, limit, querySpec.getArgs());
```

Comment:
@wmathurin do we need to set a HARD limit on query? We know that it crashes with 1000 objects. Thoughts on that?

Reply:
It really depends on what you have in the database. So app developer should use a value that makes sense for them.
If you have "reasonable" data e.g. short strings, numbers etc, then you can go a lot further than 1000. If you have "giant" JSON blobs, it's a different story.